### DISCUSSION TITLE
How to filter items in a virtual list

### DISCUSSION THREAD

**Discussion by YarivGilad** [TanStack/virtual#290] [Q&A]
Hi. 
I'm trying to find a simple example of typing some string to text input and filtering a list of items using react-virtual.
As of now, I filter the original array, while the loop runs on the **rowVirtualizer.virtualItems** array, and display the filtered item in their original position in the unfiltered list. this is buggy... what am I missing? my guess is there should be a way to "force" a re-calculation of the virtual list since now it is shorter and we need to re-render. Any ideas?

**Comment by KazimirPodolski** [comment]
How about this:
* `estimateSize` should return 0 if an item with given index is filtered out
* call `measure` when input value changes
* hide filtered out items with condition or `display: none` or anything

**Reply by piecyk** [reply]
@YarivGilad are you using correct keys via `getItemKey` https://tanstack.com/virtual/v3/docs/api/virtualizer#getitemkey also if you use dynamic, to force measurement each re-render you can use the inline hack, `ref={el => virtualRow.measureElement(el)}`
